src/process_util.rs

```diff
@@ -47,16 +47,33 @@ pub fn squash_stdio(cmd: &mut Command) -> Result<(), String> {
     Ok(())
 }
 
-pub fn capture_stdout(mut cmd: Command) -> Result<Vec<u8>, String> {
+pub struct CapturedOutput {
+    pub stdout: Vec<u8>,
+    pub stderr: Vec<u8>,
+}
+
+impl CapturedOutput {
+    fn new() -> Self {
+        Self {
+            stdout: Vec::with_capacity(1024),
+            stderr: Vec::with_capacity(1024),
+        }
+    }
+}
+
+pub fn capture_output(mut cmd: Command) -> Result<CapturedOutput, String> {
     cmd.stdout(Stdio::piped());
+    cmd.stderr(Stdio::piped());
     info!("Running {}", cmd_show(&cmd));
     let mut child = cmd.spawn()
         .map_err(|e| format!("{}", e))?;
-    let mut buf = Vec::with_capacity(1024);
-    child.stdout.take().unwrap().read_to_end(&mut buf)
-        .map_err(|e| format!("Error reading from pipe: {}", e))?;
+    let mut output = CapturedOutput::new();
+    child.stdout.take().unwrap().read_to_end(&mut output.stdout)
```

Comment:
The problem with this is that it's potentially blocking. I.e. if process have written more than (in-kernel) buffer size into stderr, and it needs to write to stderr before it will write and close stdout this will be a problem.

On the other hand, current Rust has `Command::output()` method that does the job. So this function is barely needed for better logging and more context for errors.

Reply:
Oh, I see it now. It is really tricky case.
But here we use `unshare::Command`. Stdlib's `Command::output()` uses under the hood a private [read2](https://doc.rust-lang.org/src/std/sys/unix/pipe.rs.html#81) function.

Possible there should be an analogue functionality inside `unshare`. Or is it worth to make 2 functions for capturing `stdout` and `stderr`?

Reply:
Oh, I entirely forgot that we don't have that in unshare :(

Yes two functions is fine. Enum as you have done is fine too.